scripts/imgtool/keys/general.py

```diff
@@ -30,6 +32,25 @@ def emit_c_public(self, file=sys.stdout):
                 len_format="const unsigned int {}_pub_key_len = {{}};".format(self.shortname()),
                 file=file)
 
+    def emit_raw_public(self, file=sys.stdout):
+        sys.stdout.close = lambda: None
+        with (open(file, 'wb') if file and file is not sys.stdout else sys.stdout.buffer) as file:
+            file.write(self.get_public_bytes())
+
+    def emit_c_public_hash(self, file=sys.stdout):
+        self._emit(
+                header="const unsigned char {}_pub_hash_key[] = {{".format(self.shortname()),
+                trailer="};",
+                encoded_bytes=self.get_public_hash_bytes(),
```

Comment:
Currently, the `get_public_hash_bytes()` is implemented for RSA only - as the commit message also says.
For other key types, Python throws an error... even if it's not implemented for other keys (but it probably wouldn't be too much work) it would be better to handle this error with an _unsupported...() class method.
I think it is even better to remove `get_public_hash_bytes()` and calculate the hash value of the public key (returned by the given key type's get_public_bytes() method) in general.py as part of the `emit_...._public_hash()` functions. I would suggest the latter.

Reply:
Removed the `get_public_hash_bytes() ` from the RSA class and moved it to the `KeyClass` instead as suggested.

Reply:
Going to update the commit message as well.